internal/core/src/storage/OpenDALChunkManager.cpp

```diff
@@ -0,0 +1,204 @@
+        storageType = "oss";
+    } else if (storage_config.cloud_provider == "azure") {
+        storageType = "azblob";
+    } else {
+        storageType = "s3";
+    }
+
+    op_options_ = opendal_operator_options_new();
+    if (!storage_config.access_key_id.empty() &&
+        !storage_config.access_key_value.empty()) {
+        opendal_operator_options_set(
+            op_options_, "access_key_id", storage_config.access_key_id.c_str());
+        opendal_operator_options_set(op_options_,
+                                     "secret_access_key",
+                                     storage_config.access_key_value.c_str());
+        storageType = "s3";
+    }
+    opendal_operator_options_set(op_options_, "root", "/");
+    opendal_operator_options_set(
+        op_options_, "bucket", storage_config.bucket_name.c_str());
+    opendal_operator_options_set(op_options_,
+                                 "region",
+                                 storage_config.region.empty()
+                                     ? "us-east-1"
+                                     : storage_config.region.c_str());
+    opendal_operator_options_set(
+        op_options_,
+        "endpoint",
+        ((storage_config.useSSL ? "https://" : "http://") +
+         storage_config.address)
+            .c_str());
+    opendal_operator_options_set(
+        op_options_,
+        "enable_virtual_host_style",
+        storage_config.useVirtualHost ? "true" : "false");
+
+    op_ptr_ = opendal_operator_new(storageType.c_str(), op_options_);
+    AssertInfo(op_ptr_ != nullptr, "init opendal operator failed");
+    LOG_SEGCORE_INFO_ << "init OpenDALChunkManager with parameter[storage: '"
+                      << storageType << ", " << storage_config.cloud_provider
+                      << "', endpoint: '" << storage_config.address
+                      << "', default_bucket_name:'"
+                      << storage_config.bucket_name << "', use_secure:'"
+                      << std::boolalpha << storage_config.useSSL << "']";
+}
+
+OpenDALChunkManager::~OpenDALChunkManager() {
+    opendal_operator_free(op_ptr_);
+    opendal_operator_options_free(op_options_);
+}
+
+uint64_t
+OpenDALChunkManager::Size(const std::string& filepath) {
+    return GetObjectSize(default_bucket_name_, filepath);
+}
+
+bool
+OpenDALChunkManager::Exist(const std::string& filepath) {
+    return ObjectExists(default_bucket_name_, filepath);
+}
+
+void
+OpenDALChunkManager::Remove(const std::string& filepath) {
+    DeleteObject(default_bucket_name_, filepath);
+}
+
+std::vector<std::string>
+OpenDALChunkManager::ListWithPrefix(const std::string& filepath) {
+    return ListObjects(default_bucket_name_.c_str(), filepath.c_str());
+}
+
+uint64_t
+OpenDALChunkManager::Read(const std::string& filepath,
+                          void* buf,
+                          uint64_t size) {
+    if (!ObjectExists(default_bucket_name_, filepath)) {
```

Comment:
should not check exist here, it will introduce an extra rpc.

GetObjectBuffer should return some error message if Object not exist

Reply:
We should also change the implementation in MinioChunkManager too

Reply:
opened an issue to tracking that https://github.com/apache/incubator-opendal/issues/3257